### PR TITLE
Resolves client-side duplicate event receival in case of client side initiated invalidations

### PR DESCRIPTION
**Pull request by ahmetmircik**
Resolves client-side duplicate event receival in case of client side initiated invalidations

closes https://github.com/hazelcast/hazelcast/issues/7194

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/client/MapAddNearCacheEntryListenerRequest.java [hunk 1]
```diff
@@ -61,12 +67,46 @@ protected Object newMapListener(ClientEndpoint endpoint) {
...
+                }
+            } else if (!endpoint.getUuid().equals(event.getSourceUuid())) {
+                return event;
+            }
+
+            return null;
+        }
+
+        protected List<SingleNearCacheInvalidation> excludeThisSourcesInvalidations(BatchNearCacheInvalidation event) {
+            List<SingleNearCacheInvalidation> invalidations = event.getInvalidations();
```
Comment (serkan-ozal) [inline hazelcast/src/main/java/com/hazelcast/map/impl/client/MapAddNearCacheEntryListenerRequest.java]: As far as I understood, this `BatchNearCacheInvalidation` can be sent to multiple `InvalidationListener` for different endpoints. So this means that it is shared. But in here we are modifying its included `SingleNearCacheInvalidation`s by removing them through iterator. So this effects others and removed `SingleNearCacheInvalidation` would be used by another `InvalidationListener` for different endpoint. 

Am I missing something here?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/client/MapAddNearCacheEntryListenerRequest.java]: Fixed

**Comment by serkan-ozal** [comment]
This should also fix https://github.com/hazelcast/hazelcast/issues/7194

**Comment by jerrinot** [comment]
@serkan-ozal: Have you had a chance to fully review this changeset?

**Comment by serkan-ozal** [comment]
:+1:

**Comment by gurbuzali** [comment]
:+1:

**Comment by jerrinot** [comment]
@ahmetmircik: is backport needed here?

**Pull request merged**